### ISSUE TITLE
Cannot disable folding to avoid conflicts

### ISSUE THREAD

**Issue by lyndhurst** [SidOfc/mkdx#114]
# Github issue

**OS type**: Unix - Neon 5.19 (focal based)

**Vim**: version 8.1.2269

**Reproduce steps**:

1. Plugin settings in `.vimrc` are the default 'getting started' settings from the README, with any of the following variations:
    1. `'fold': { 'enable': 1 , 'components': ['toc']}`
    2. `'fold': { 'enable': 1 }`
    3. `'fold': { 'enable': 0 }`
    4. No `'fold'` setting at all
2. Set `g:markdown_folding = 1` . I do not have installed nor use [tpope's plugin](https://github.com/tpope/vim-markdown).
3. I tried with my usual markdown folding plugin from [masukomi](https://github.com/masukomi/vim-markdown-folding) with the same result as the default builtin folding method.
3. Open a markdown document

**Expected**:

mkdx will do nothing with `foldmethod` and `foldexpr` and leave them intact for whatever plugin I use for folding, or the builtin vim fold method. _As I understood from the README section on folds._

**Actual**:

A warning message appears at the bottom saying:

> Pattern not found: mkdx#settings.fold.enable = 0

The plugin also interferes with [masukomi's vim-markdown-folding](https://github.com/masukomi/vim-markdown-folding) as I do not get the same folding results with and without mkdx.

Maybe I understood your instructions incorrectly, and, mkdx cannot be used with another folding method; or only with [tpope's plugin](https://github.com/tpope/vim-markdown). I hope not because some other mkdx features are actually pretty neat.

Thanks for your time.

**Comment by lyndhurst** [comment]
After trying out some of the features, I find many features being kind of unstable. 

I installed with `vim-plug` the same way as described in the instructions, but I noticed that mkdx gets reinstalled every single time I run `PlugInstall` and often removed when I run `PlugClean` without any change on my config.

Maybe the problems are related...

**Comment by SidOfc** [comment]
Hi @lyndhurst, thank you for reporting the issue in such detail, I will look into it as soon as I can, would you mind sharing your vimrc file with me via gist or a repo link?

As for the `vim-plug` issue, unfortunately I can't reproduce it, for me running `:PlugInstall` does not reinstall mkdx if it's already there, likewise `:PlugClean` does not remove it. I'm not sure how long it was since you last installed vim-plug, could you perhaps try giving it an update?

I'm also assuming you've found issues with other features as well, could you summarize which and what was wrong?

**Comment by lyndhurst** [comment]
Hi,
I tried again this morning after reading your request for my [vimrc file](https://gist.github.com/lyndhurst/20d2ccbd16a0c714f9f7dc33796b7fa0). So I cheated a little, and cleaned up said file before trying to reproduce the errors, here is what I can and cannot reproduce with that config:

- I do not get this annoying error message about some fold pattern not found,
- mkdx does not seem to interfere with the folding anymore (whichever method I try)
- `PlugClean` does not randomly remove your plugin anymore
- `PlugInstall` still re-installs it consistently
- I am not getting any link concealing.

About the other "unstabilities" I was experiencing, it was mostly unconsistent concealing. This morning testing convinced me that something was wrong in my config.

I am sorry, I think I opened an issue too fast, I will try to test more thoroughly starting with a clean config. Unfortunately I will not have time to do it right away. In the meantime, you should be able to reproduce the `PlugInstall` issue with that gist; I tested it on a clean VM with the latest Kde neon iso.

Thank you for your interest in that issue, I will get back to you if I have better information on that folding error.

**Comment by SidOfc** [comment]
Hey @lyndhurst cheers for checking this out! I'll try to reproduce this re-install issue with the gist later tonight and see if I can figure it out, no rush about testing with a clean config, do it when you're comfortable and have some time I'd say :+1: 

I'll leave this issue open until I've at least looked into this re-install issue and will report back when I know more.
Cheers for the feedback so far!

**Comment by lyndhurst** [comment]
Sorry, I was away for the weekend.

I can confirm that I am not getting any error from the `let g:markdown_folding = 1`, and it works fine.

Unfortunately, I still get no concealing and this re-install problem is still there. I tested with an even cleaner config and removed all custom theming to be sure. As you noticed, I had a very 'hacky' highlighting fix for a `vim-notes` plugin I was trying out.

Seeing how thorough you are in resolving this issue, I feel like I should test again as soon as I have more time, because there has to be something wrong on my side. I am in the process of re installing my whole system, and as soon as I can I will start another VM, and try again with nothing installed but a very minimalist vim config.

Thank you for taking the time to help me out so meticulously. I will get you some feedback as soon as I can be more reactive.
Cheers.

**Comment by SidOfc** [comment]
No problem, hope you enjoyed it :)

Good to know the `let g:markdown_folding = 1` is fixed, that was actually a syntax error on my part. I might try installing KDE Neon in VirtualBox for myself to check if there are any issues, cheers for pointing out the distro :+1:

I've got one quite "out there" idea which we haven't explored yet, but are the permissions on and inside the `~/.vim/plugged` folder set to your username?

![image](https://user-images.githubusercontent.com/3225058/91025494-c0a6dd80-e5f9-11ea-946b-0b9a3abfb3ed.png)

Just grasping at straws here at the moment... don't stress too much about the testing as I'll likely look at this myself as well. This re-install issue is a weird one, not to mention not getting any link concealing!

Cheers for your time and attention to these issues as well!

**Comment by lyndhurst** [comment]
Ok, so I tried again to check your permissions idea, and I have no idea what changed between now, and a couple of hours ago, but, everything is working fine now; concealing, folding, and I have no problem with `PlugInstall` neither.

I did not change anything to the config file, maybe some residual files, or config from another plugin when I was switching from one config to another. I was usually restarting vim after running `PlugClean` and `PlugInstall`.

I do not feel like I did anything different this time. But I am glad I tested right away hopefully saving you some useless headaches :smile: .

Anyway thanks a lot for your help, I think you can close this.

**Comment by SidOfc** [comment]
Sounds good, once again thanks for checking it out with me, much appreciated :+1: 

If issues start popping up again feel free to reopen this (if related) or open a new issue, cheers!
